Make it easier to use DataGrip w/ Synapse's schema (#14982)
matrix-org/synapse: contrib/datagrip/README.md

# Schema symlinks

This directory contains symlinks to the latest dump of the postgres full schema. This is useful to have, as it allows IDEs to understand our schema and provide autocomplete, linters, inspections, etc.

> I (dmr) use the DataGrip functionality in IntelliJ's paid-for products. It seems to only consider files called `*.sql` when defining a schema from DDL; `*.sql.postgres` wasn't detected for me.

Reviewer: ```suggestion
When using the DataGrip functionality in IntelliJ's products. It seems to only consider files called `*.sql` when defining a schema from DDL; `*.sql.postgres` will be ignored.
```

What's the ramification of this?

Author: Not sure what you're getting at. Some context:

My first attempt to get this working was to point the IDE at the directories `synapse/storage/[logical db]/full_schemas/72` and let it scan those files there. Unfortunately it only seems to look for files ending in `.sql`; it doesn't read the `.sql.postgres` nor `.sql.sqlite` files at all. I couldn't find a way to point the IDE at specific files either.

Hence my approach here: make symlinks to the postgres schema dumps ending in `.sql`, and place the links in one directory. (I deliberately chose to ignore SQLite.)

Will update the comment to better explain this.

Reviewer: Maybe something like:

```suggestion
When using the DataGrip functionality in IntelliJ's products. It seems to only consider files called `*.sql` when defining a schema from DDL; `*.sql.postgres` will be ignored. To get around this we symlink those files to ones ending in `.sql`.
```

Author: See 1f1b5b068a

## Example
![](datagrip-aware-of-schema.png)

## Caveats

- Doesn't include temporary tables created ad-hoc by Synapse.
- Postgres only. IDEs will likely be confused by SQLite-specific queries.

> - Will not include migrations created after the latest schema dump.

Reviewer: I guess we could symlink each of these individually? That seems like overkill though...
